packages/devtools_app/lib/src/debugger/codeview.dart

```diff
@@ -66,8 +67,10 @@ class _CodeViewState extends State<CodeView> with AutoDisposeMixin {
   }
 
   void _parseScriptLines() {
-    // Parse the source into lines.
-    lines = script.source?.split('\n') ?? [];
+    // Create a new SyntaxHighlighter with the script's source in preparation
+    // for building the code view.
+    highlighter = SyntaxHighlighter(source: script?.source ?? '');
```

Comment:
Do we pay the cost of parsing the source here? I assume that we highlight the whole file and don't do something like lazily highlight sections.

If we highlight the whole file, we might want a cut-out for very large files. dart:html for instance is (from memory) 1.5MB. I'd assume we wouldn't be able to highlight that performantly. You might want to do a quick check for very large files and add an arbitrary cut-off for large files.

Reply:
No, we incur that cost at [line 219](https://github.com/flutter/devtools/pull/2583/files#diff-486524c0f38b9588774d5a6c3942a7d646cbd82e8398047ed0a0a541b7b84e1dR219) since we can't generate the `TextSpan`s without a `BuildContext`. You're correct: we do highlight the whole file at this point since doing anything lazy here would probably require a complete overhaul of `CodeView`.

I just looked into seeing if there was performance issues analyzing `dart:html`, but it looks like it's not included in the library list when inspecting DevTools on web (is that expected?). A cut-off sounds like a reasonable idea, but we should probably figure out what "very large" should be defined as.

Reply:
Ah, `dart:html` would likely only be visible if debugging a flutter web / dart web app.

You can try `package:flutter/src/material/icons.dart`; I'm not sure how it compares to dart:html, but it is ~17k lines line / 1.1MB.

Reply:
That's what I was doing. I used DevTools on web to self-inspect and didn't see `dart:html`. That seems strange to me.

Anyway, I went ahead and opened the icons file using DevTools for desktop and it did take a few seconds to load (I'm sure it'll be much worse on web). I'll use the character count of that file to come up with some sort of upper limit.

Reply:
I gave some more thought about how to parse the script asynchronously so we could avoid introducing an upper limit but I think we'd need to push that work onto another thread, which isn't exactly trivial at this point with no isolate support on web. I've set an upper limit of 500,000 characters for now and added a TODO to revisit this later, potentially when Flutter Web has `compute` support.